bin/p2-log-bridge/main.go

```diff
@@ -20,31 +22,56 @@ func main() {
 	kingpin.Version(version.VERSION)
 	kingpin.Parse()
 
+	if err := setSTDINToBlock(); err != nil {
+		logging.DefaultLogger.WithError(err).Error("fatal error setting STDIN to block")
+		os.Exit(1)
+	}
+
+	var wg sync.WaitGroup
 	loggerCmd := exec.Command(*durableLogger)
 	durablePipe, err := loggerCmd.StdinPipe()
 	if err != nil {
-		logging.DefaultLogger.WithError(err).Errorln("failed during logbridge setup")
-		os.Exit(1)
+		logging.DefaultLogger.WithError(err).Error("fatal error during configuration of subordinate log command")
 	}
 
-	go handleSignals(durablePipe)
+	wg.Add(1)
+	go func(loggerCmd exec.Cmd, wg sync.WaitGroup) {
+		defer wg.Done()
 
-	go func(r io.Reader, durableWriter io.Writer, nonDurableWriter io.Writer, logger logging.Logger) {
-		logbridge.Tee(r, durableWriter, nonDurableWriter, logger)
-	}(os.Stdin, durablePipe, os.Stdout, logging.DefaultLogger)
+		if err := loggerCmd.Start(); err != nil {
+			logging.DefaultLogger.WithError(err).Errorln("fatal error during execution of subordinate log command")
+			os.Exit(1)
+		}
 
-	loggerCmd.Start()
-	err = loggerCmd.Wait()
-	if err != nil {
-		logging.DefaultLogger.WithError(err)
-		os.Exit(1)
-	}
+		err := loggerCmd.Wait()
+		if err != nil {
+			logging.DefaultLogger.WithError(err).Errorln("fatal error in subordinate log command")
+			os.Exit(1)
+		}
+	}(*loggerCmd, wg)
+
+	logbridge.Tee(os.Stdin, durablePipe, os.Stdout, logging.DefaultLogger)
```

Comment:
The command has two outputs formats—one lossy, one not—and it has two output channels—one via a command, one through stdout—so perhaps it doesn't matter much how those two pairs match up.